content/guides/getstarted/setup-local-development.md

```diff
@@ -38,48 +38,266 @@ deploying Solana programs to the blockchain.
-your local system:
+Just because rust compiles and builds your software into binary that can run for the computer architecture we specify, we need to install some OS level dependencies on our machine. 
 
-```bash
-sh -c "$(curl -sSfL https://release.solana.com/stable/install)"
+### Let’s start with Windows 8/10/11
+
+There are two ways to get started on windows, 
+
+1.  build on windows
+2.  build on windows but with WSL
+
+WSL is windows subsystem for linux, which translates Linux system calls into Windows system calls using a compatibility layer, **for running Linux binary executables natively** and it runs Linux processes alongside the Windows kernel in a lightweight virtual machine. 
+
+WSL and linux will mostly have same steps to go through.
+
+### Build Dependencies for Windows
+
+1. Rust requires build tools to compile for our Solana Tools.
+Download the build tools from here - 
+choose C/C++ build tools by minimum
+
+When you download, make sure to run the tool as an administrator (right click on executable and choose run as administrator)
+    
+    [Download Visual Studio Tools - Install Free for Windows, Mac, Linux](https://visualstudio.microsoft.com/downloads/?q=build+tools#build-tools-for-visual-studio-2022)
+    
+2. Now Open a Command Prompt (`cmd.exe`) as an Administrator
+    
+    ![setup—-windows.exe](/assets/guides/setup-local-environment/setup---rustup-windows.png)
+    
+---
+
+### Build Dependencies for WSL
+
+1. Install WSL in your system, make sure to restart your computer when installing is done.
+    
+    [Install WSL](https://learn.microsoft.com/en-us/windows/wsl/install)
+    
+2. Follow this tutorial to open VSCode in WSL 
+    
+    [](https://code.visualstudio.com/docs/remote/wsl-tutorial)
+    
+3. Install dependencies required, make sure you open VSCode as an administrator before switching to WSL. Then open a terminal inside VSCode and type this command to install build essential dependencies. Rust and Anchor might need these. 
+    
+    ```jsx
+    sudo apt-get install \
+        build-essential \
+        pkg-config \
+        libudev-dev llvm libclang-dev \
+        protobuf-compiler libssl-dev
+    ```
+    
+4. done with installing dependencies on WSL, Continue to install rust in the next section.
+
+---
+
+### Build Dependencies for Linux
+
+Installing dependencies in linux is same as windows with WSL.
+
+```jsx
+sudo apt-get install -y \
+    build-essential \
+    pkg-config \
+    libudev-dev llvm libclang-dev \
+    protobuf-compiler libssl-dev
+```
+
+---
+
+### Build Dependencies for macOS
+
+In macOS build tools are given by Xcode command line tools. You can download it from this link. 
+Sign in with your apple Id and download. 
+
+You can check if your xcode cli is installed via this command.
+
+```jsx
+xcode-select -p
+```
+
+If you don’t see a path returned, you need to install the cli tools.
+
+There are three ways to install xcode cli tools
+
+1. Install tools via terminal
+    
+    ```jsx
```

Comment:
Again this is `bash` (it's probably zsh on a mac but `bash` syntax highlighting works)